From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NzSN <aydenlin@foxmail.com>
Date: Tue, 5 Sep 2023 11:40:20 +0800
Subject: Logging debug info to exec path instead of CWD

CWD may varing which is make it harder to found by users.

diff --git a/content/app/content_main.cc b/content/app/content_main.cc
index d34480ca23849f98a36b3cb31add6a07e375cf82..71879a465c91bbe55a6545f0d1a2dd6d2275e62a 100644
--- a/content/app/content_main.cc
+++ b/content/app/content_main.cc
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include <fstream>
 #include "content/public/app/content_main.h"
 
 #include "base/allocator/buildflags.h"
@@ -49,6 +50,7 @@
 
 #if BUILDFLAG(IS_WIN)
 #include <windows.h>
+#include <libloaderapi.h>
 
 #include "base/win/process_startup_helper.h"
 #include "base/win/win_util.h"
@@ -345,10 +347,13 @@ void NO_STACK_PROTECTOR CMDSwitchSetup() {
 
   base::CommandLine::ForCurrentProcess()->AppendSwitch(::switches::kEnableLogging);  
 
-  // Setup logfile path to CWD.
-  WCHAR path[MAX_PATH];
-  ::GetCurrentDirectory(MAX_PATH, path);
-  base::FilePath logPath{path};
+  // Setup logfile path to where executable residein.
+  WCHAR execPath[MAX_PATH];
+  GetModuleFileNameW(NULL, execPath, MAX_PATH);
+  std::wstring execDir{execPath};
+  execDir = (execDir.substr(0, execDir.find_last_of(L"\\/")) + L"\\debug.log");  
+  base::FilePath logPath{execDir.c_str()};
+
   base::CommandLine::ForCurrentProcess()->AppendSwitchPath(::switches::kLogFile, logPath);
 }
 
